### PR TITLE
[HYDRATOR-1131] CSVParser to add entry in error dataset in case of unformatted record in csv

### PR DESCRIPTION
**Pull request by shashankmoghe**

### CODE REVIEW

**Comment by shashankmoghe** [comment]
@abhinavbansal246 Please review

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) th
         }
       }
     } catch (IOException e) {
-      throw Throwables.propagate(e);
+      emitter.emitError(new InvalidEntry<>(31, e.getStackTrace()[0].toString() + " : " + e.getMessage(), in));
```
Comment (abhinavbansal246) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java:208]: Is 31 standard error code for invalid entry for csv?

**Comment by abhinavbansal246** [comment]
Can we add a test to see if the errors are being written to the configured error dataset?
Please update the description of the delimiter field. Add that in case of null it defaults to ","

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java (hunk 1 above)
Comment (shashankmoghe) [reply transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java:208]: This is  not something which is standard for CSV. I referred one of the exiting implementation of emitError in XMLParser. Could you please suggest the error code format ?

**Comment by shashankmoghe** [comment]
I have already added a test for emitError. Please review.

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java [hunk 2]
```diff
@@ -260,7 +260,7 @@ private StructuredRecord createStructuredRecord(@Nullable CSVRecord record, Stru
 
     @Nullable
     @Description("Custom delimiter to be used for parsing the fields. The custom delimiter can only be specified by " +
-      "selecting the option 'Custom' from the format drop-down.")
+      "selecting the option 'Custom' from the format drop-down.In case of null, defaults to ','.")
```
Comment (russorat) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java]: need a space between . and In

**Comment by abhinavbansal246** [comment]
What happens when the error dataset is not configured? Does the plugin throws error in such a scenario if it receives malformed row while parsing.

**Comment by abhinavbansal246** [comment]
Can we update description accordingly. conveying that the errored rows will be committed to the error dataset if configured or the plugin will fail otherwise (if that is the case)

**Comment by shashankmoghe** [comment]
If error dataset is configured then all the errored rows, if present in the CSV, will be committed
 to the specified error dataset. If not configured, the errored rows will be committed to default error dataset.

**Comment by abhinavbansal246** [comment]
LGTM
@shashankmoghe paste ITN results

**Comment by shashankmoghe** [comment]
ITN result:
[INFO] Reactor Summary:
[INFO]
[INFO] integration-test .................................. SUCCESS [2.648s]
[INFO] integration-test-core ............................. SUCCESS [3.014s]
[INFO] integration-test-remote ........................... SUCCESS [5:16.484s]
[INFO] integration-test-standalone ....................... SUCCESS [1.006s]
[INFO] upgrade-test ...................................... SUCCESS [1.577s]
[INFO] long-running-test ................................. SUCCESS [1.368s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5:26.206s
[INFO] Finished at: Tue Dec 13 14:29:32 IST 2016
[INFO] Final Memory: 79M/333M
[INFO] ------------------------------------------------------------------------

**Pull request merged**